assembly/src/parsers/u32_ops.rs

```diff
@@ -543,22 +535,21 @@ pub fn parse_u32lt(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Asse
 /// Specifically we test the first two numbers to be u32, then perform a gt check and flip the
 /// results.
 ///
-/// TODO: add VM cycle count.
+/// VM cycles per mode:
+/// u32lte: 13 cycles
+/// u32lte.unsafe: 7 cycles
 pub fn parse_u32lte(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
-    match op.num_parts() {
-        0 => return Err(AssemblyError::missing_param(op)),
-        1 => {
-            // make sure the first two values on the stack are u32 (while sapping their order)
-            assert_u32_operands(span_ops, false);
+    handle_u32_and_unsafe_check(span_ops, op, false)?;
+    if op.num_parts() == 2 {
+        // Swap the order in unsafe mode since we only swap on u32 check.
+        span_ops.push(Operation::Swap);
+    }
```

Comment:
Couldn't we just pass *true* as the last parameter to `handle_u32_and_unsafe_check()` and avoid this *if* statement?

Reply:
We can't as we special case unsafe and basically do nothing in that method.

Reply:
ah - I see. Basically, in case of `unsafe` the order doesn't change, but in case of `safe` it does, right? I think this flow makes it a bit difficult to follow. I wonder if it would be better if `handle_u32_and_unsafe_check()` returned a boolean which indicated whether the order was flipped or not. This way, it will be easy to understand why we need another swap here.